proxy/proxy.go

```diff
@@ -870,6 +870,11 @@ func (p *Proxy) makeBackendRequest(ctx *context, requestContext stdlibcontext.Co
 	req = injectClientTrace(req, ctx.proxySpan)
 
 	response, err := roundTripper.RoundTrip(req)
+	if response.StatusCode >= 400 {
```

Comment:
this we can't do on >=400. We need to check the errors that return timeouts and connection errors. 
Everything that is a valid response we do accept as "success".

Reply:
I agree with the fact error code >=400 is not what we want because 4xx is a valid response, its client side error. But 5xx even being a valid response means something is up with the backend, 5xx are always server side errors. So maybe something like `err != nil || esponse.StatusCode >= 500` would be better?